robot_localization:
  ekf_node:
    ros__parameters:
      odom_frame: "odom"
      base_link_frame: "camera_optical"
      world_frame: "earth"
      map_frame: "earth"

      #publish_tf: true

      frequency: 10.0

      # need predict_to_current_time if required output Hz is higher than input
      #predict_to_current_time: true
      sensor_timeout: 10.0
      #pose0_queue_size: 20
      pose0_rejection_threshold: 2.0

      dynamic_process_noise_covariance: True
      transform_timeout: 0.200

      use_sim_time: false
      two_d_mode: false

      #smooth_lagged_data: true
      #history_length: 10.0
      #print_diagnostics: true

      # Fuse absolute pose estimated from map rasters
      # TODO: enable orientation fusion
      pose0: "/gisnav/pose_node/pose"
      pose0_config: [true, true, true,     # Position XYZ
                     false, false, false, #true, true, true,     # Orientation (roll, pitch, yaw)
                     false, false, false,  # Velocity XYZ
                     false, false, false,  # Angular rates XYZ
                     false, false, false]  # Accelerations XYZ

      # Fuse smooth relative pose estimated via VO differentially as velocity
      pose1: "/gisnav/twist_node/pose"
      pose1_differential: true
      pose1_config: [true, true, true,     # Position XYZ
                     true, true, true,     # Orientation (roll, pitch, yaw)
                     false, false, false,  # Velocity XYZ
                     false, false, false,  # Angular rates XYZ
                     false, false, false]  # Accelerations XYZ

      #debug: true
      #debug_out_file: "/home/hmakelin/robot_localization_debug.txt"
